From 950cc9d750e6a978ffcc13f45cc53a3649b47be8 Mon Sep 17 00:00:00 2001
From: JS Deck <jsdeckerido@gmail.com>
Date: Sun, 7 Dec 2025 02:55:09 -0400
Subject: [PATCH 6/9] setupapi: SetupDiGetDeviceRegistryProperty now returns
 all lower case for SPDRP_BASE_CONTAINERID; SetupDiGetDeviceRegistryPropertyA
 should return wide string for type GUID.

---
 dlls/setupapi/devinst.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/dlls/setupapi/devinst.c b/dlls/setupapi/devinst.c
index e74dccb353b..f99d80fdf02 100644
--- a/dlls/setupapi/devinst.c
+++ b/dlls/setupapi/devinst.c
@@ -3696,7 +3696,16 @@ BOOL WINAPI SetupDiGetDeviceRegistryPropertyA(HDEVINFO devinfo,
     if (Property < ARRAY_SIZE(PropertyMap) && PropertyMap[Property].nameA)
     {
         DWORD size = PropertyBufferSize;
-        LONG l = RegQueryValueExA(device->key, PropertyMap[Property].nameA,
+        LONG l;
+
+        /* Despite GUID returned as REG_SZ, it's always wide string on Windows. */
+        if (PropertyMap[Property].devPropType == DEVPROP_TYPE_GUID)
+            return SetupDiGetDeviceRegistryPropertyW(
+                devinfo, device_data, Property, PropertyRegDataType,
+                PropertyBuffer, PropertyBufferSize, RequiredSize
+            );
+
+        l = RegQueryValueExA(device->key, PropertyMap[Property].nameA,
                 NULL, PropertyRegDataType, PropertyBuffer, &size);
 
         if (l == ERROR_FILE_NOT_FOUND)
@@ -3710,6 +3719,7 @@ BOOL WINAPI SetupDiGetDeviceRegistryPropertyA(HDEVINFO devinfo,
         if (RequiredSize)
             *RequiredSize = size;
     }
+
     return ret;
 }
 
@@ -3751,7 +3761,12 @@ BOOL WINAPI SetupDiGetDeviceRegistryPropertyW(HDEVINFO devinfo,
             SetLastError(l);
         if (RequiredSize)
             *RequiredSize = size;
+
+        /* Death Stranding Director's Cut expect the containerId string to be all lower case. */
+        if (ret && PropertyMap[Property].devPropType == DEVPROP_TYPE_GUID)
+            _wcslwr((LPWSTR)PropertyBuffer);
     }
+
     return ret;
 }
 
-- 
2.52.0

